oauthproxy.go

```diff
@@ -687,11 +692,31 @@ func (p *OAuthProxy) OAuthStart(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	var codeChallenge, codeVerifier, codeChallengeMethod string
+	if p.provider.Data().CodeChallengeMethod != "" {
+		codeChallengeMethod = p.provider.Data().CodeChallengeMethod
+		codeVerifier, err = generateRandomString(128)
```

Comment:
This might be best align to other random patterns & URL safety by using our other methods which generate random `[]bytes` and base64 URL safe encode it. 96 bytes will base64 encode to exactly 128 characters if yo need that (3->4 expansion in size in base64 encoding).

Reply:
See section 4.1 and 4.2 about the constraints for the code verifier: https://datatracker.ietf.org/doc/html/rfc7636#section-4.1

I'm looking at your `Nonce()` code in `pkg/encryption` but it does not have the same rune limitations that I need. Unfortunately.

Reply:
I think that link specifies that the technique to Base64 + URLEncode (a-zA-Z0-9-_) will follow the spec:

> It is RECOMMENDED that the output of
>    a suitable random number generator be used to create a 32-octet
>    sequence.  The octet sequence is then base64url-encoded to produce a
>    43-octet URL safe string to use as the code verifier.

I only recommend it since that pattern (random []bytes, + base64 to get a URL safe string) is used throughout the codebase & the implementation (and speed) is much cleaner than iterating through a list of legal characters to select 1 by 1.

Albeit, your use case likely merits extending `Nonce()` to be more customizable if you want to go larger than the 32bytes of entropy it has by default today for oauth state & oidc nonce usages.